spyder/plugins/editor/plugin.py

```diff
@@ -757,30 +751,34 @@ def on_completions_available(self):
         widget = self.get_widget()
         completions = self.get_plugin(Plugins.Completions)
 
+        widget.sig_after_configuration_update_requested.connect(
+            self._after_configuration_update
+        )
         self.sig_file_opened_closed_or_updated.connect(
             completions.file_opened_closed_or_updated
         )
 
         completions.sig_language_completions_available.connect(
             widget.register_completion_capabilities)
         completions.sig_open_file.connect(widget.load)
-        completions.sig_editor_rpc.connect(widget._rpc_call)
         completions.sig_stop_completions.connect(
             widget.stop_completion_services)
 
     @on_plugin_teardown(plugin=Plugins.Completions)
     def on_completions_teardown(self):
         widget = self.get_widget()
         completions = self.get_plugin(Plugins.Completions)
+
+        widget.sig_after_configuration_update_requested.disconnect(
+            self._after_configuration_update
```

Comment:
```suggestion
            completions.after_configuration_update
```

To go along with my previous suggestion.